src/helpers/buffers/fsv.rs

```diff
@@ -38,62 +28,93 @@ impl<const N: usize> FixedSizeByteVec<N> {
         Self {
             data: vec![0_u8; N * capacity],
             added: bitvec![0; capacity],
-            taken: 0,
+            capacity,
```

Comment:
maybe it is a good time to reject `capacity = 0`? does not have to be in this change though